src/api/transactions/trade.ts

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+"use strict";
+
+import { Transaction, registerTransactionType } from "./transaction";
+import { Address } from "#erdstall/ledger";
+import * as assets from "#erdstall/ledger/assets";
+import { jsonObject, jsonMember } from "typedjson";
+import { ABIEncoder, ABIPacked, BigInteger } from "#erdstall/api/util";
+import { Signature } from "#erdstall/api";
+import { Signer, utils } from "ethers";
+
+@jsonObject
+export class TradeFees {
+	constructor(market: Address, fee: assets.Assets) {
+		this.market = market;
+		this.fee = fee;
+	}
+	@jsonMember(Address) market: Address;
+	@jsonMember(()=>assets.Assets) fee: assets.Assets;
+
+	asABITagged(contract: Address): ABIPacked {
+		return new ABIEncoder(this.market, this.fee).pack("ErdstallTradeFees", contract);
+	}
+}
+
+const tradeTypeName = "Trade";
+
+@jsonObject
+export class TradeOffer {
+	@jsonMember(Address) owner: Address;
+	@jsonMember(()=>assets.Assets) offer: assets.Assets;
+	@jsonMember(()=>assets.Assets) request: assets.Assets;
+	@jsonMember(BigInteger) expiry: BigInteger;
+	@jsonMember(TradeFees) fees?: TradeFees;
+	@jsonMember(Signature) sig?: Signature;
+
+	constructor(
+		owner: Address,
+		offer: assets.Assets,
+		request: assets.Assets)
+	{
+		this.owner = owner;
+		this.offer = offer;
+		this.request = request;
+		this.expiry = new BigInteger(2n**64n - 1n); // For now, never expire.
+		this.fees = undefined; // For now, ignore fees.
+		this.sig = undefined;
```

Comment:
I think you can just remove those two lines and these fields will be `undefined`.